lib/pirates/game_server/instance.ex

```diff
@@ -87,42 +87,43 @@ defmodule Pirates.GameServer.Instance do
     # init timer
     :timer.send_interval(@tick_timer_in_ms, :tick)
 
-    {:ok, table}
+    {:ok, {name, table}}
   end
 
-  def handle_call(:register, {from_pid, _}, table) do
+  def handle_call(:register, {from_pid, _}, {name, table}) do
     if :ets.insert_new(table, {from_pid, %{}}) do
       Process.monitor(from_pid)
       IO.puts "Just registered #{inspect(from_pid)}"
-      {:reply, {:ok, table}, table}
+      {:reply, {:ok, table}, {name, table}}
     else
-      {:reply, {:error, "#{inspect(from_pid)} is already registered on this server"}, table}
+      {:reply, {:error, "#{inspect(from_pid)} is already registered on this server"}, {name, table}}
     end
   end
 
-  def handle_call(:table, _from, table) do
-    {:reply, table, table}
+  def handle_call(:table, _from, {name, table}) do
+    {:reply, table, {name, table}}
   end
 
-  def handle_info(:tick, table) do
+  def handle_info(:tick, {name, table}) do
     {pids, states} = Pirates.GameServer.Instance.state(table)
-    Enum.each(pids, fn pid -> send(pid, {:state_tick, states}) end)
-    {:noreply, table}
+    # Enum.each(pids, fn pid -> send(pid, {:state_tick, states}) end)
+    Pirates.Endpoint.broadcast("game:" <> name, "state_tick", %{state: states})
```

Comment:
This should be https://hexdocs.pm/phoenix/Phoenix.Channel.html#broadcast_from/3 I think. The `Endpoint.broadcast` is a callback, I don't think you're actually supposed to manually call it. Additionally, the `broadcast_from` will automatically filter out the calling socket, so no need to do the handle_out intercepting later

But much cleaner, thanks! You can delete the previous line that's commented out

Reply:
Nah it's fine to use Endpoint broadcast if you don't want to hold reference to the socket (which I don't want Instance to do because every client will have one so it's ambiguous which we should hold).

See 
https://hexdocs.pm/phoenix/Phoenix.Channel.html#module-broadcasting-to-an-external-topic

Also the handle_out isn't filtering the socket to send to, but rather the player state for each individual client channel (i.e. player 4 will get all states except for player 4's state).

Reply:
Oh, I see. Thanks for the explanation!